src/kvstore.c

```diff
@@ -90,6 +95,25 @@ typedef struct {
     kvstore *kvs;
 } kvstoreHashtableMetadata;
 
+/**********************************/
+/*** Static Helpers ***************/
+/**********************************/
+
+/* A dict type for just storing integers as keys, without values. */
+static uint64_t dictIntHash(const void *key) {
+    /* We hash the pointer value itself. */
+    return dictGenHashFunction(&key, sizeof(key));
+}
+
+static int dictIntCompare(const void *key1, const void *key2) {
+    return key1 == key2;
+}
+
+static dictType kvstoreImportingDictType = {
+    .hashFunction = dictIntHash,
+    .keyCompare = dictIntCompare,
+};
```

Comment:
Minor: You could use `hashtable` instead of `dict` for this set of integers. It's more efficient and compact since doesn't use a dictEntry. The integers (cast as pointers) would be stored directly in the big array. Not sure if it matters much if this is not a hot path.